src/main/java/com/yahoo/sketches/pig/tuple/ArrayOfDoublesSketchesToPValueEstimates.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017, Yahoo! Inc.
+ * Licensed under the terms of the Apache License 2.0. See LICENSE file at the project root for terms.
+ */
+
+package com.yahoo.sketches.pig.tuple;
+
+import com.yahoo.memory.Memory;
+import com.yahoo.sketches.tuple.ArrayOfDoublesSketch;
+import com.yahoo.sketches.tuple.ArrayOfDoublesSketches;
+import com.yahoo.sketches.tuple.ArrayOfDoublesSketchIterator;
+
+import java.io.IOException;
+
+import org.apache.commons.math3.stat.inference.TTest;
+import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
+
+import org.apache.pig.EvalFunc;
+import org.apache.pig.data.DataByteArray;
+import org.apache.pig.data.Tuple;
+
+/**
+ * Calculate p-values given two ArrayOfDoublesSketch. Each value in the sketch
+ * is treated as a separate metric measurement, and a p-value will be generated
+ * for each metric.
+ */
+public class ArrayOfDoublesSketchesToPValueEstimates extends EvalFunc<Tuple> {
+    @Override
+    public Tuple exec(final Tuple input) throws IOException {
+        if ((input == null) || (input.size() != 2)) {
+            return null;
+        }
+
+        // Get the two sketches
+        final DataByteArray dbaA = (DataByteArray) input.get(0);
+        final DataByteArray dbaB = (DataByteArray) input.get(1);
+        final ArrayOfDoublesSketch sketchA = ArrayOfDoublesSketches.wrapSketch(Memory.wrap(dbaA.get()));
+        final ArrayOfDoublesSketch sketchB = ArrayOfDoublesSketches.wrapSketch(Memory.wrap(dbaB.get()));
+
+        // Check that the size of the arrays in the sketches are the same
+        if (sketchA.getNumValues() != sketchB.getNumValues()) {
+            throw new IllegalArgumentException("Both sketches must have the same number of values");
+        }
+
+        // Store the number of metrics
+        int numMetrics = sketchA.getNumValues();
+
+        // Check if either sketch is empty
+        if (sketchA.isEmpty() || sketchB.isEmpty()) {
+            return null;
+        }
+
+        // If the sketches contain fewer than 2 values, the p-value can't be calculated
+        if (sketchA.getRetainedEntries() < 2 || sketchB.getRetainedEntries() < 2) {
+            return null;
```

Comment:
On both this line and line 50, I am not sure the value that should be returned when one of the sketches is empty or a sketch has fewer than 2 values. An exception doesn't make sense, as they are valid sketch objects. We could return a p-value array of all `1.0`, as we can't establish that the values are different. Returning a null value instead of an array seems like it might be problematic in some cases. Thoughts @AlexanderSaydakov @jmalkin?